.github/workflows/ci.yml

```diff
@@ -107,3 +121,61 @@ jobs:
           poetry install
           poetry run agbenchmark start --mock
           poetry run agbenchmark start --mock --maintain
+          poetry build
+
+      - name: Run regression tests
+        run: |
+          cd agent/$AGENT_NAME
+          if [ "$AGENT_NAME" == "gpt-engineer" ]; then
+            make install
+            source venv/bin/activate
+          elif [ "$AGENT_NAME" == "Auto-GPT" ]; then
+            python -m venv venv
+            source venv/bin/activate
+            pip install -r requirements.txt
+          elif [ "$AGENT_NAME" == "mini-agi" ]; then
+            python -m venv venv
+            source venv/bin/activate
+            pip install -r requirements.txt
+            cp .env_example .env
+          elif [ "$AGENT_NAME" == "smol-developer" ]; then
+            python -m venv venv
+            source venv/bin/activate
+            pip install -r requirements.txt
+          elif [ "$AGENT_NAME" == "SuperAGI" ]; then
+            cp config_template.yaml config.yaml
+            sed -i 's/OPENAI_API_KEY:.*/OPENAI_API_KEY: "'"${{ secrets.OPENAI_API_KEY }}"'"/' config.yaml
+            docker-compose up -d --build
+          else
+            echo "Unknown agent name: $AGENT_NAME"
+            exit 1
+          fi
```

Comment:
I will put these into a file called: 
setup_auto-gpt.sh

and then dynamically call the right setup depending on the agent.